adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationActivityUnitTest.java

```diff
@@ -173,26 +174,32 @@ public void testReturnToCaller() throws IllegalArgumentException, NoSuchFieldExc
      * Return authentication exception at setResult so that mActivity receives at
      * onActivityResult
      */
-    @SmallTest
-    @UiThreadTest
-    public void testWebviewAuthenticationException() throws IllegalArgumentException,
-            NoSuchFieldException, IllegalAccessException, InvocationTargetException,
-            ClassNotFoundException, NoSuchMethodException, InstantiationException,
-            InterruptedException, ExecutionException {
-        startActivity(mIntentToStartActivity, null, null);
-        mActivity = getActivity();
+    @Test
+    public void testWebviewAuthenticationException() throws Throwable {
+        mActivityRule.launchActivity(mIntentToStartActivity);
         AuthenticationSettings.INSTANCE.setDeviceCertificateProxyClass(MockDeviceCertProxy.class);
         MockDeviceCertProxy.reset();
         MockDeviceCertProxy.setIsValidIssuer(true);
         MockDeviceCertProxy.setPrivateKey(null);
-        String url = AuthenticationConstants.Broker.PKEYAUTH_REDIRECT
+        final String url = AuthenticationConstants.Broker.PKEYAUTH_REDIRECT
                 + "?Nonce=nonce1234&CertAuthorities=ABC&Version=1.0&SubmitUrl=submiturl&Context=serverContext";
-        WebViewClient client = getCustomWebViewClient();
-        WebView mockview = new WebView(getActivity().getApplicationContext());
-        ReflectionUtils.setFieldValue(mActivity, "mSpinner", null);
+        final WebViewClient client = getCustomWebViewClient();
```

Comment:
Since there're 2 intents starting the Activity among the test case class, I could not launch Activity in the 'setUp' (with @Before) method. What I could do is only launching Activity in separate test methods. Previously, there's the "@UiThreadTest" annotation on the method. But I have to remove it since it would crash on launching Activity. Then, there're some methods must be called on UI thread, then I have to use 'mActivityRule.runOnUiThread'. Yes, there's some copy/paste on code that must run on UI thread. thanks